### DISCUSSION TITLE
[Q&A] Keeping history of inventory from FusionInventory 9.5 to GLPI-agent v1.4

### DISCUSSION THREAD

**Discussion by Jibe-2** [glpi-project/glpi-agent#377] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I wanted to know, if it exist a method to keep my previous FusionInventory history from a linux VM with GLPI 9.5.12 to transfer on my new VM GLPI 10.0.7 and transfer the history of each computer to GLPI-agent v1.4.

**Comment by g-bougard** [comment]
Hi @Toriiebay 

are you talking about the asset history in GLPI ? If yes, this is not an agent question and the history of an asset is kept during an upgrade. I'm not aware of any tool to transfer an asset history from a GLPI instance to another.

**Reply by Jibe-2** [reply]
Not the asset history, When you go to the path : Asset > Computers > "(random PC)" > History

I want to keep this history of modification, through the years and so transfer from FusionInventory to GLPI-agent

Ex: I have a computer "PC-22000" with FusionInventory-agent with its modification and its history
when i install the new GLPI-agent on the "PC-22000" i want to transfer the history of the fusioninventory-agent to the GLPI-agent

I hope to be clear

**Reply by g-bougard** [reply]
No sorry, this is not clear.

So you want to keep a history of modification... but of what ? Share screenshots if this help you to show what you want to keep: the actual history you're talking about in GLPI 9.5 or with FI agent and the one in GLPI 10 or GLPI Agent.

**Reply by Jibe-2** [reply]
(Sorry for my english, i'm french ^^)
I try to explain more simply.

Currently, we updated GLPI 9.5 to 10.0.7 and transfer all data from FusionInventory-agent to GLPI 10.0.7.
Now, we want to know if it's possible for GLPI-agent 1.4 can take over from FusionInventory 2.6 automatically and if yes, how ?

**Reply by g-bougard** [reply]
FI Agent & GLPI Agent are like friends as GLPI Agent has been forked from FI Agent (and as I was the last active maintainer of FI Agent). They are just behave the same way. The inventory format integration is done the same way in GLPI (even if GLPI Agent decides to use JSON as new format). So as you replace FI Agent by GLPI Agent, you'll just see GLPI Agent will replace FI Agent as inventory agent on the asset.

On windows, you can use the official VBS which can try to uninstall FI Agent before installing GLPI Agent. On other platforms, just uninstall FI Agent before installing GLPI Agent.
Virtually, you still can have the 2 agents running on a platform but only one agent needs to be running. If you have 2 agents running against the same GLPI target you'll just see automatic agent switching and maybe some changes in the inventory as GLPI Agent has many more fixes than FI Agent.

Don't forget to follow all cautions on that documentation to be sure to use the right server URL target: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

I hope I finally understand well your question and you have the expected answer.

**Reply by Jibe-2** [reply]
The VBS is very usefull but the problem is not resolved.

1) Script executed
2) FIA uninstalled 
3) GLPI-agent installed
4) On GLPI-10 we have both agent (FIA & GLPI-agent) for the same PC but GLPI-agent does not seem to be related to the computer because the rules do not apply (screen of VDB-2844).
![GLPI](https://user-images.githubusercontent.com/108300308/232748369-c7ff45fb-28b9-4ee0-8f61-2ad92c1b61bb.png)

**Reply by g-bougard** [reply]
It seems GLPI-Agent inventory is not imported for any reason. You should check Glpi-Agent log, eventually enabling `debug = 2` mode. You should also probably verify the `php-errors.log` GLPI file for any error while the inventory is submitted. You may also have to check if update & import rules are corrects.

**Reply by Jibe-2** [reply]
After we have checked in the error files, we found this : 
![glpi-agent log](https://user-images.githubusercontent.com/108300308/232774428-d83bb569-6c79-4607-a578-46036dc2b174.png) 
![error log](https://user-images.githubusercontent.com/108300308/232774432-f35938ba-92cb-410c-9fbd-a654d0198658.png)

Do you have some information about this ?

**Reply by g-bougard** [reply]
Only the full stack of the "PHP Fatal error" can tell us what's going on. But just in case, you still check the database schema using the following command:
https://glpi-install.readthedocs.io/en/latest/command-line.html#database-schema-check

About the "Can't decode JSON content" error, this is a normal error as Deploy & Collect task can only be used with GlpiInventory plugin or FI plugin. If you need these tasks, you should probably install GlpiInventory from the marketplace and update the server URL accordingly. If you don't need them, you may disable that tasks on the agent or just not install them with the agent.

**Reply by Jibe-2** [reply]
Thanks you, all right now !
